fdbserver/MoveKeys.actor.cpp

```diff
@@ -1454,6 +1598,27 @@ ACTOR static Future<Void> startMoveShards(Database occ,
 						    .detail("DestID", destId)
 						    .detail("ReadVersion", tr.getReadVersion().get());
 
+						// Pre validate consistency of update of keyServers and serverKeys
+						if (SERVER_KNOBS->AUDIT_DATAMOVE_PRE_CHECK &&
+						    auditStorageFailedCount < SERVER_KNOBS->AUDIT_DATAMOVE_MAX_SUCCESSIVE_FAILED) {
+							state std::vector<Future<Void>> validationActors;
+							state std::unordered_map<UID, bool> validationResult;
+							for (auto& ssid : src) {
+								validationActors.push_back(
+								    store(validationResult[ssid],
+								          validateRangeAssignment(
+								              &tr, rangeIntersectKeys, ssid, "startMoveShards_precheck")));
+							}
+							for (auto& ssid : dest) {
+								validationActors.push_back(
+								    store(validationResult[ssid],
+								          validateRangeAssignment(
+								              &tr, rangeIntersectKeys, ssid, "startMoveShards_precheck")));
+							}
+							wait(waitForAll(validationActors));
```

Comment:
How are we going to use the validationResult? Will we pause DD once we find any issue?